cmd/clusterctl/cmd/root.go

```diff
@@ -48,12 +54,33 @@ func Execute() {
 func init() {
 	flag.CommandLine = flag.NewFlagSet(os.Args[0], flag.ExitOnError)
 
-	verbosity := flag.CommandLine.Int("v", 0, "Set the log level verbosity.")
-	logf.SetLogger(logf.NewLogger(logf.WithThreshold(verbosity)))
+	verbosity = flag.CommandLine.Int("v", 0, "Set the log level verbosity. This overrides the CLUSTERCTL_LOG_LEVEL environment variable.")
 
 	RootCmd.PersistentFlags().AddGoFlagSet(flag.CommandLine)
 	RootCmd.PersistentFlags().StringVar(&cfgFile, "config", "",
 		"Path to clusterctl configuration (default is `$HOME/.cluster-api/clusterctl.yaml`)")
+
+	cobra.OnInitialize(initConfig)
+}
+
+func initConfig() {
+	// check if the CLUSTERCTL_LOG_LEVEL was set via env var or in the config file
+	if *verbosity == 0 {
+		configClient, err := config.New(cfgFile)
+		if err == nil {
+			v, err := configClient.Variables().Get("CLUSTERCTL_LOG_LEVEL")
+			if err == nil && v != "" {
+				verbosityFromEnv, err := strconv.Atoi(v)
+				if err != nil {
+					fmt.Printf("Failed to convert CLUSTERCTL_LOG_LEVEL string to an int. err=%s", err.Error())
```

Comment:
Add a new line at the end and redirect to `os.Stderr`.
```suggestion
					fmt.Fprintf(os.Stderr, "Failed to convert CLUSTERCTL_LOG_LEVEL string to an int. err=%s\n", err.Error())
```

Reply:
done